Modules/_elementtree.c

```diff
@@ -865,7 +875,12 @@ _elementtree_Element___deepcopy___impl(ElementObject *self, PyObject *memo)
         }
 
         assert(!element->extra->length);
-        element->extra->length = self->extra->length;
+        /*
+         * The original 'self->extra' may be gone at this point if deepcopy()
+         * had side-effects. However, 'i' is the number of copied items that
+         * we were able to successfully copy.
+         */
+        element->extra->length = i;
```

Comment:
Not directly related but the `Py_REFCNT(object) == 1` in `deepcopy()` below is likely unsafe for similar reasons as #132070. I don't really understand what that check is doing, though: it seems what it does could work too if the refcnt > 1.

Reply:
Oh I don't know. I'm leaving on Monday evening for ~10 days. Maybe we can track the issue separately?

Reply:
Yes definitely a separate issue! Just noticed it while reading code around your change.

Reply:
If the refcnt > 1, we need to check in memo, and use a reference to already made copy if it is found. We need also to update memo. It is complicated and adds a lot of overhead.